Source/Core/AudioCommon/CubebStream.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "AudioCommon/CubebStream.h"
+#include "AudioCommon/DPL2Decoder.h"
+#include "Common/Logging/Log.h"
+#include "Common/Thread.h"
+#include "Core/ConfigManager.h"
+
+#include "cubeb/cubeb.h"
+
+constexpr u32 BUFFER_SAMPLES = 512;  // ~10 ms - needs to be at least 240 for surround
+
+long CubebStream::data_callback(cubeb_stream* stream, void* user_data, const void* /*inputbuffer*/,
+                                void* outputbuffer, long nframes)
+{
+  auto self = static_cast<CubebStream*>(user_data);
+
+  if (self->m_stereo)
+  {
+    self->m_mixer->Mix(static_cast<short*>(outputbuffer), nframes);
+  }
+  else
+  {
+    if (static_cast<size_t>(nframes) * 2 > self->m_conversion_buffers_size)
```

Comment:
This could probably be simplified by using `std::vector` instead of `std::unique_ptr<[]>` and just calling `std::vector::reserve` unconditionally.

I don't have very strong opinions on this though.